datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -52,26 +44,81 @@ use crate::{
     physical_plan::{planner::DefaultPhysicalPlanner, ColumnarValue, PhysicalExpr},
 };
 
+/// Interface to pass statistics information to [`PruningPredicates`]
```

Comment:
Here is the alternate proposed API based off of @Dandandan  and @jorgecarleitao 's comments here: https://github.com/apache/arrow-datafusion/pull/380#discussion_r638143765